### PR TITLE
avoiding the buffer hack and using a flag instead

### PR DESCRIPTION
**Pull request by shireenf-ibm**
#118

### CODE REVIEW

**Comment by shireenf-ibm** [comment]
I followed the solution in #118, 
However by following the instructions [here](https://github.com/IBM/network-config-analyzer/blob/master/tests/test_rest_api.md) I got the expected results on both master branch and the current one.

**Comment by zivnevo** [comment]
Try to directly use the flag in `NetworkConfig.__init__()` rather than putting it into a data member and using it later.

File: network-config-analyzer/NetworkConfig.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def __init__(self, name, peer_container, entry_list=None, config_type=None):
         self.referenced_ip_blocks = None
         self.type = config_type or NetworkConfig.ConfigType.Unknown
         self.allowed_labels = set()
+        if from_buffer:
+            entry_list = 'buffer: ' + entry_list
         peer_container.clear_pods_extra_labels()
         for entry in entry_list or []:
             self.add_policies_from_entry(entry)
```
Comment (zivnevo) [inline network-config-analyzer/NetworkConfig.py:58]:
```suggestion
        peer_container.clear_pods_extra_labels()
        if from_buffer:
            entry_list = 'buffer: ' + entry_list
        else:
            for entry in entry_list or []:
                self.add_policies_from_entry(entry)
```
Sorry, I wasn't clear enough. Here is my suggestion.

**Review by zivnevo — CHANGES_REQUESTED** [review]
Sorry, I wasn't clear enough. Here is my suggestion.

File: network-config-analyzer/NetworkConfig.py (hunk 1 above)
Comment (zivnevo) [inline network-config-analyzer/NetworkConfig.py:58]:
```suggestion
        peer_container.clear_pods_extra_labels()
        if from_buffer:
            self._add_policies(entry, 'buffer', True)
        else:
            for entry in entry_list or []:
                self.add_policies_from_entry(entry)
```

**Review by zivnevo — CHANGES_REQUESTED** [review]
Corrected suggestion

**Comment by zivnevo** [comment]
ok, so this fails too. How about using two different parameters to `NetworkConfig.__init__()`?
That is, something like:
```
    def __init__(self, name, peer_container, entry_list=None, config_type=None, buffer=None):
```
And then, call the ctor from `RestServer.py` with `entry_list` set to `None` and `buffer` set to `entry`.

**Comment by shireenf-ibm** [comment]
> ok, so this fails too. How about using two different parameters to `NetworkConfig.__init__()`? That is, something like:
> 
> ```
>     def __init__(self, name, peer_container, entry_list=None, config_type=None, buffer=None):
> ```
> 
> And then, call the ctor from `RestServer.py` with `entry_list` set to `None` and `buffer` set to `entry`.

working :)

**Review by zivnevo — APPROVED** [review]
hooray

**Pull request merged**